functions/submit-finding.js

```diff
@@ -107,46 +147,69 @@ exports.handler = async (event) => {
     if (!confirmed) {
       return {
         statusCode: 401,
-        body: "Unauthorized",
+        body: JSON.stringify({
+          error: "Authorization failed",
+        }),
       };
     }
-    if (attributedTo !== user) {
+    if (isTeamSubmission) {
       const url = `${event.headers.origin}/.netlify/functions/get-user?id=${attributedTo}`;
       const response = await fetch(url);
       if (!response.ok) {
         return {
           statusCode: 401,
-          body: "Unauthorized",
+          body: JSON.stringify({
+            error: "You must be registered to submit findings",
+          }),
         };
       }
       const team = await response.json();
       if (!team || !team.members || !team.members.includes(user)) {
         return {
           statusCode: 401,
-          body: "Unauthorized",
+          body: JSON.stringify({
+            error: "You cannot submit findings for a team you are not on",
+          }),
         };
       }
+      if (!team.address) {
+        // create a PR to update team JSON file with team address
+        try {
+          await updateTeamData(team, address);
+        } catch (error) {
+          // don't throw error if this PR fails - there will likely be duplicates
```

Comment:
This.. worries me.. but good notes to keep around for later review :)

Reply:
Fair, but there are no real consequences to this failing. It's just trying to save the team's address for future submissions. I figured we wouldn't want to block a submission for this, even if it was a different failure. But I could at least check that the failure is due to a branch with that name already existing, and throw an error for any other failures.